cmp/grid/columns/Column.js

```diff
@@ -239,26 +247,34 @@ export class Column {
                         {value, data: record} = agParams;
                     return elementRenderer(value, {record, agParams, column: me});
                 }
+
                 refresh() {return false}
             };
         }
 
-        // Support enhanced, absValue-aware sorting via GridSorters in GridModel.sortBy[].
         const sortCfg = find(gridModel.sortBy, {colId: ret.colId});
         if (sortCfg) {
             ret.sort = sortCfg.sort;
             ret.sortedAt = gridModel.sortBy.indexOf(sortCfg);
+        }
 
+        // Support enhanced, absValue-aware sorting via GridSorters in GridModel.sortBy[].
+        if (this.isTreeColumn) {
             // ag-Grid sort impl. sources its primary values from the node's `groupData` property,
             // which is not what we want when sorting treeColumns.
-            if (this.isTreeColumn) {
-                ret.comparator = (v1, v2, node1, node2) => {
-                    return sortCfg.comparator(node1.data[field], node2.data[field]);
-                };
-            } else {
-                ret.comparator = (v1, v2) => sortCfg.comparator(v1, v2);
-            }
+            ret.comparator = (v1, v2, node1, node2) => {
+                const sortCfg = find(gridModel.sortBy, {colId: ret.colId});
+                if (sortCfg) return sortCfg.comparator(node1.data[field], node2.data[field]);
```

Comment:
Think this code might be easier to reason about if it were a single comparator, that just reset its v1 and v2 at the top, if it was a TreeColumn